AWS/connectors/s3/__init__.py

```diff
@@ -162,15 +163,19 @@ async def next_batch(self, previous_processing_end: float | None = None) -> tupl
                         async with self.s3_wrapper.read_key(bucket=s3_bucket, key=normalized_key) as content:
                             records.extend(self._parse_content(self.decompress_content(content)))
 
+                        if len(records) >= self.limit_of_events_to_push:
+                            continue_receiving = False
+                            result += await self.push_data_to_intakes(events=records)
+
                     except Exception as e:
                         self.log(
                             message=f"Failed to fetch content of {record}: {str(e)}",
                             level="warning",
                         )
 
-            if len(records) >= self.limit_of_events_to_push or not records:
+            if not records:
                 continue_receiving = False
 
-        result = await self.push_data_to_intakes(events=records)
+        result += len(await self.push_data_to_intakes(events=records))
```

Comment:
Maybe we can add a condition to check if `records` is empty or not and avoid calling `push_data_to_intakes` for empty list.